projects/admin/src/app/circulation/patron/change-password-form/change-password-form.component.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2020 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { FormlyFieldConfig } from '@ngx-formly/core';
+import { TranslateService } from '@ngx-translate/core';
+import { BsModalRef, BsModalService } from 'ngx-bootstrap/modal';
+import { ToastrService } from 'ngx-toastr';
+import { User } from '../../../class/user';
+import { UserService } from '../../../service/user.service';
+
+@Component({
+  selector: 'admin-change-password-form',
+  templateUrl: './change-password-form.component.html'
+})
+export class ChangePasswordFormComponent implements OnInit {
+
+  /** patron to change the password */
+  patron: User;
+
+  /** form */
+  form: FormGroup = new FormGroup({});
+
+  /** model */
+  model = {};
+
+  /** form fields */
+  formFields: FormlyFieldConfig[];
+
+  /**
+   * Constructor
+   * @param _modalService - BsModalService
+   * @param _bsModalRef - BsModalRef
+   * @param _translateService - TranslateService
+   * @param _toastr - ToastrService
+   * @param _userService - UserService
+   */
+  constructor(
+    private _modalService: BsModalService,
+    private _bsModalRef: BsModalRef,
+    private _translateService: TranslateService,
+    private _toastr: ToastrService,
+    private _userService: UserService
+  ) { }
+
+  /**
+   * Component initialization.
+   */
+  ngOnInit() {
+    const initialState: any = this._modalService.config.initialState;
+    if (initialState.hasOwnProperty('patron')) {
+      this.closeModal();
+    }
+    this.patron = initialState.patron;
+    this._initForm();
+  }
+
+  /**
+   * Submit form
+   * @param model - Object
+   */
+  submit(patron, model) {
+    this._userService.changePassword(patron.username, model.password).subscribe(
+      () => {
+        this._toastr.success(
+          this._translateService.instant('The patron password has been changed.'),
+        );
+        this.closeModal();
+      },
+      (resp) => {
+        console.log('Error: Update Patron Password', resp);
```

Comment:
Remove `console.log`

Reply:
I think that it is to leave at is as it is displayed only when an error occurs. This can be useful. But if you want I can remove it.